# 🎯 Module 1 – Probability Practice Questions


## ❓ Question 1
### A dice is rolled. What is the probability of getting an even number?

💡 **Explanation:**
- Total outcomes = 6  →  {1, 2, 3, 4, 5, 6}
- Even numbers = {2, 4, 6}  →  3 outcomes
- Formula:  P = Favorable / Total = 3 / 6 = 0.5

In [1]:
# Question 1 - Even number on dice

total_outcomes = 6
favorable_outcomes = 3    # {2, 4, 6}

probability = favorable_outcomes / total_outcomes

print("Total outcomes   =", total_outcomes)
print("Favorable (even) =", favorable_outcomes)
print("Probability      =", probability)

Total outcomes   = 6
Favorable (even) = 3
Probability      = 0.5


---
## ❓ Question 2
### A coin is tossed. What is the probability of getting Tail?

💡 **Explanation:**
- Sample space = {Head, Tail}  →  Total = 2
- Favorable = {Tail}  →  1
- P(Tail) = 1 / 2 = 0.5

In [2]:
# Question 2 - Tail on coin toss

total_outcomes = 2       # Head, Tail
favorable_outcomes = 1   # only Tail

probability = favorable_outcomes / total_outcomes

print("Total outcomes    =", total_outcomes)
print("Favorable (Tail)  =", favorable_outcomes)
print("P(Tail)           =", probability)

Total outcomes    = 2
Favorable (Tail)  = 1
P(Tail)           = 0.5


---
## ❓ Question 3
### A dice is rolled. What is the probability of getting a number greater than 4?

💡 **Explanation:**
- Total outcomes = 6
- Numbers greater than 4 = {5, 6}  →  2 outcomes
- P = 2 / 6 = 0.333

In [3]:
# Question 3 - Number greater than 4

total_outcomes = 6
favorable_outcomes = 2    # {5, 6}

probability = favorable_outcomes / total_outcomes

print("Total outcomes        =", total_outcomes)
print("Favorable (5 and 6)   =", favorable_outcomes)
print("P(number > 4)         =", probability)
print("P(number > 4) approx  =", round(probability, 3))

Total outcomes        = 6
Favorable (5 and 6)   = 2
P(number > 4)         = 0.3333333333333333
P(number > 4) approx  = 0.333


---
## ❓ Question 4
### P(Rain) = 0.65. What is P(No Rain)?

💡 **Explanation:**
- This is Complementary Event
- Formula:  P(A') = 1 - P(A)
- P(No Rain) = 1 - 0.65 = 0.35

In [4]:
# Question 4 - Complementary Event

p_rain = 0.65

# Formula: P(No Rain) = 1 - P(Rain)
p_no_rain = 1 - p_rain

print("P(Rain)    =", p_rain)
print("P(No Rain) = 1 -", p_rain, "=", p_no_rain)

# Check: both must add up to 1
print("Check: ", p_rain, "+", p_no_rain, "=", p_rain + p_no_rain)

P(Rain)    = 0.65
P(No Rain) = 1 - 0.65 = 0.35
Check:  0.65 + 0.35 = 1.0


---
## ❓ Question 5
### What is the probability of getting number 3 on a dice?

💡 **Explanation:**
- This is a Simple Event (only one outcome)
- Total outcomes = 6
- Favorable = {3}  →  1
- P(3) = 1 / 6

In [5]:
# Question 5 - Simple Event (getting 3 on dice)

total_outcomes = 6
favorable_outcomes = 1    # only {3}

probability = favorable_outcomes / total_outcomes

print("Total outcomes    =", total_outcomes)
print("Favorable         =", favorable_outcomes, "  (only number 3)")
print("P(3)              =", probability)
print("P(3) approx       =", round(probability, 4))

Total outcomes    = 6
Favorable         = 1   (only number 3)
P(3)              = 0.16666666666666666
P(3) approx       = 0.1667


---
## ❓ Question 6
### A coin is tossed 10 times. Head comes 6 times. What is the Empirical Probability of Head?

💡 **Explanation:**
- Empirical Probability = based on real experiment
- Formula:  P = Number of times event happened / Total trials
- P(Head) = 6 / 10 = 0.6

In [6]:
# Question 6 - Empirical Probability

total_trials = 10
head_count = 6

# Empirical Formula
p_head = head_count / total_trials

print("Total trials         =", total_trials)
print("Head appeared        =", head_count, "times")
print("Empirical P(Head)    =", p_head)

# Compare with theoretical
p_theoretical = 1 / 2
print("Theoretical P(Head)  =", p_theoretical)
print("Are they same?       =", p_head == p_theoretical)

Total trials         = 10
Head appeared        = 6 times
Empirical P(Head)    = 0.6
Theoretical P(Head)  = 0.5
Are they same?       = False


---
## ❓ Question 7
### One card is drawn from a deck of 52 cards. What is the probability it is a Heart?

💡 **Explanation:**
- Total cards = 52
- Heart cards = 13
- P(Heart) = 13 / 52 = 0.25

In [7]:
# Question 7 - Classical Probability (Cards)

total_cards = 52
heart_cards = 13

probability = heart_cards / total_cards

print("Total cards   =", total_cards)
print("Heart cards   =", heart_cards)
print("P(Heart)      =", probability)

Total cards   = 52
Heart cards   = 13
P(Heart)      = 0.25


---
## ❓ Question 8
### Dice is rolled. A = getting 2, B = getting 5. Find P(A or B).

💡 **Explanation:**
- A and B cannot happen at same time → Mutually Exclusive
- Formula:  P(A or B) = P(A) + P(B)
- P(A) = 1/6,  P(B) = 1/6
- P(A or B) = 1/6 + 1/6 = 2/6 = 0.333

In [8]:
# Question 8 - Addition Rule (Mutually Exclusive)

total_outcomes = 6

# Event A = getting 2
p_A = 1 / total_outcomes

# Event B = getting 5
p_B = 1 / total_outcomes

# A and B cannot happen together (mutually exclusive)
# Formula: P(A or B) = P(A) + P(B)
p_A_or_B = p_A + p_B

print("P(A) = P(getting 2) =", p_A)
print("P(B) = P(getting 5) =", p_B)
print("P(A or B)           =", p_A, "+", p_B, "=", p_A_or_B)
print("P(A or B) approx    =", round(p_A_or_B, 4))

P(A) = P(getting 2) = 0.16666666666666666
P(B) = P(getting 5) = 0.16666666666666666
P(A or B)           = 0.16666666666666666 + 0.16666666666666666 = 0.3333333333333333
P(A or B) approx    = 0.3333


---
## ❓ Question 9
### Card drawn from deck. A = Heart, B = King. Find P(A or B).

💡 **Explanation:**
- A and B CAN happen together (King of Heart exists) → NOT Mutually Exclusive
- Formula:  P(A or B) = P(A) + P(B) - P(A and B)
- P(A) = 13/52,  P(B) = 4/52,  P(A and B) = 1/52
- P(A or B) = 13/52 + 4/52 - 1/52 = 16/52

In [9]:
# Question 9 - Addition Rule (Not Mutually Exclusive)

total_cards = 52

# Event A = Heart card
heart_cards = 13
p_A = heart_cards / total_cards

# Event B = King card
king_cards = 4
p_B = king_cards / total_cards

# A and B together = King of Heart (only 1 card)
p_A_and_B = 1 / total_cards

# Formula: P(A or B) = P(A) + P(B) - P(A and B)
p_A_or_B = p_A + p_B - p_A_and_B

print("P(A) = P(Heart) =", heart_cards, "/", total_cards, "=", p_A)
print("P(B) = P(King)  =", king_cards, "/", total_cards, "=", p_B)
print("P(A and B)      = 1 /", total_cards, "=", p_A_and_B)
print("")
print("P(A or B) = P(A) + P(B) - P(A and B)")
print("P(A or B) =", p_A, "+", p_B, "-", p_A_and_B)
print("P(A or B) =", p_A_or_B)
print("P(A or B) = 16/52 =", round(p_A_or_B, 4))

P(A) = P(Heart) = 13 / 52 = 0.25
P(B) = P(King)  = 4 / 52 = 0.07692307692307693
P(A and B)      = 1 / 52 = 0.019230769230769232

P(A or B) = P(A) + P(B) - P(A and B)
P(A or B) = 0.25 + 0.07692307692307693 - 0.019230769230769232
P(A or B) = 0.3076923076923077
P(A or B) = 16/52 = 0.3077


---
## ❓ Question 10
### Check the 3 Axioms of Probability using a coin toss.

💡 **Explanation:**
- Axiom 1: P(A) >= 0  →  No probability is negative
- Axiom 2: P(S) = 1   →  All probabilities add up to 1
- Axiom 3: P(Head or Tail) = P(Head) + P(Tail)  →  for mutually exclusive

In [10]:
# Question 10 - 3 Axioms of Probability

p_head = 0.5
p_tail = 0.5

print("=== Coin Toss ===")
print("P(Head) =", p_head)
print("P(Tail) =", p_tail)
print("")

# Axiom 1: Probability is always >= 0
print("--- Axiom 1: P(A) >= 0 ---")
print("P(Head) >= 0 :", p_head >= 0)
print("P(Tail) >= 0 :", p_tail >= 0)
print("")

# Axiom 2: Sum of all probabilities = 1
print("--- Axiom 2: P(S) = 1 ---")
total = p_head + p_tail
print("P(Head) + P(Tail) =", p_head, "+", p_tail, "=", total)
print("Sum = 1 ?", total == 1)
print("")

# Axiom 3: Mutually exclusive events
print("--- Axiom 3: P(Head or Tail) = P(Head) + P(Tail) ---")
# Head and Tail cannot happen together
p_head_or_tail = p_head + p_tail
print("P(Head or Tail) =", p_head_or_tail)
print("")

print("=== All 3 Axioms Verified! ===")

=== Coin Toss ===
P(Head) = 0.5
P(Tail) = 0.5

--- Axiom 1: P(A) >= 0 ---
P(Head) >= 0 : True
P(Tail) >= 0 : True

--- Axiom 2: P(S) = 1 ---
P(Head) + P(Tail) = 0.5 + 0.5 = 1.0
Sum = 1 ? True

--- Axiom 3: P(Head or Tail) = P(Head) + P(Tail) ---
P(Head or Tail) = 1.0

=== All 3 Axioms Verified! ===


---
## 📋 Summary of All Questions

| No. | Question | Formula Used | Answer |
|-----|----------|--------------|--------|
| 1 | P(Even on dice) | Favorable / Total | 3/6 = 0.5 |
| 2 | P(Tail on coin) | Favorable / Total | 1/2 = 0.5 |
| 3 | P(Number > 4 on dice) | Favorable / Total | 2/6 = 0.33 |
| 4 | P(No Rain) if P(Rain)=0.65 | 1 - P(A) | 0.35 |
| 5 | P(Getting 3 on dice) | Favorable / Total | 1/6 = 0.166 |
| 6 | Head 6 times in 10 trials | Times happened / Total trials | 6/10 = 0.6 |
| 7 | P(Heart from deck) | Favorable / Total | 13/52 = 0.25 |
| 8 | P(2 or 5 on dice) | P(A) + P(B) | 2/6 = 0.33 |
| 9 | P(Heart or King from deck) | P(A) + P(B) - P(A and B) | 16/52 = 0.307 |
| 10 | Verify 3 Axioms | All 3 rules | All True ✅ |

---
